experiment/coverage_utils.py

```diff
@@ -191,65 +200,49 @@ def get_coverage_infomation(coverage_summary_file):
 def store_coverage_data(experiment_config: dict):
     """Generates the specific coverage data and store in cloud bucket."""
     logger.info('Start storing coverage data')
-    with multiprocessing.Pool() as pool, multiprocessing.Manager() as manager:
-        q = manager.Queue()  # pytype: disable=attribute-error
-        covered_regions = get_all_covered_regions(experiment_config, pool, q)
-        json_src_dir = reporter.get_reports_dir()
-        filesystem.recreate_directory(json_src_dir)
-        json_src = os.path.join(json_src_dir, 'covered_regions.json')
-        with open(json_src, 'w') as src_file:
-            json.dump(covered_regions, src_file)
-        json_dst = exp_path.filestore(json_src)
-        filestore_utils.cp(json_src, json_dst)
+    with multiprocessing.Pool() as pool:
+        store_all_covered_regions(experiment_config, pool)
 
     logger.info('Finished storing coverage data')
 
 
-def get_all_covered_regions(experiment_config: dict, pool, q) -> dict:
-    """Gets regions covered for each pair for fuzzer and benchmark."""
-    logger.info('Measuring all fuzzer-benchmark pairs for final coverage data.')
+def store_all_covered_regions(experiment_config: dict, pool):
+    """Stores regions covered for each pair for fuzzer and benchmark."""
+    logger.info('Storing all fuzzer-benchmark pairs for final coverage data.')
 
     benchmarks = experiment_config['benchmarks'].split(',')
     fuzzers = experiment_config['fuzzers'].split(',')
     experiment = experiment_config['experiment']
 
-    get_covered_region_args = [(experiment, fuzzer, benchmark, q)
-                               for fuzzer in fuzzers
-                               for benchmark in benchmarks]
+    store_covered_region_args = [(experiment, fuzzer, benchmark)
+                                 for fuzzer in fuzzers
+                                 for benchmark in benchmarks]
 
-    result = pool.starmap_async(get_covered_region, get_covered_region_args)
+    pool.starmap(store_covered_region, store_covered_region_args)
+    pool.close()
```

Comment:
Do you know if close and join are actually needed?
I imagine they aren't.

Reply:
I thought that without close and join the program will proceed when the `store_covered_region` is still running, which is bad. Or am I wrong about this?

Reply:
Oh my bad. I just looked into the doc and I guess I'm wrong about this :( I'll delete it.

Reply:
I think that may have been the case when this used async.
Try this snippet 
```
import multiprocessing

pool = multiprocessing.Pool()

pool.starmap(print, [(i,) for i in range(1000)])

print('done')
```
done gets printed

Reply:
Yeah you are right.